### PR TITLE
Improve: break the list of 'with type' module constraints

### PR DESCRIPTION
**Pull request by gpetiot**
Should address #621 
I think we always want this new behavior, hence no new option

### CODE REVIEW

File: test/passing/js_source.ml.ref [hunk 1]
```diff
@@ -3649,20 +3649,25 @@ end
...
+     and type data = string
+     and type map = SSMap.map)
 ;;
 
 let ssmap :
-    (module MapT with type key = string and type data = string and type map = SSMap.map)
-    =
+    (module MapT with type key = string
+                  and type data = string
+                  and type map = SSMap.map) =
```
Comment (gpetiot) [inline test/passing/js_source.ml.ref]: So far no success breaking before `with` without introducing more regressions elsewhere.

Comment (Julow) [reply test/passing/js_source.ml.ref]: What kind of regressions ?

Comment (gpetiot) [reply test/passing/js_source.ml.ref]: You can look at the diff of my last commit 2d68308, there are net improvements (even if we break more lines sometimes) but for some cases I'm not sure if it's really better

**Comment by gpetiot** [comment]
Any thoughts on this? Can I merge it?

**Pull request merged**